apps/web/src/app/(dashboard)/documents/[id]/edit/document-edit-page-view.tsx

```diff
@@ -42,10 +46,48 @@ export const DocumentEditPageView = async ({ params, team }: DocumentEditPageVie
     teamId: team?.id,
   }).catch(() => null);
 
+  const documentVisibility = document?.visibility;
+  const currentTeamMemberRole = team?.currentTeamMember?.role;
+  const isRecipient = document?.Recipient.find((recipient) => recipient.email === user.email);
+  let canAccessDocument = true;
+
+  if (!isRecipient) {
+    canAccessDocument = match([documentVisibility, currentTeamMemberRole])
+      .with([DocumentVisibility.EVERYONE, TeamMemberRole.ADMIN], () => true)
+      .with([DocumentVisibility.EVERYONE, TeamMemberRole.MANAGER], () => true)
+      .with([DocumentVisibility.EVERYONE, TeamMemberRole.MEMBER], () => true)
+      .with([DocumentVisibility.MANAGERANDABOVE, TeamMemberRole.ADMIN], () => true)
+      .with([DocumentVisibility.MANAGERANDABOVE, TeamMemberRole.MANAGER], () => true)
+      .with([DocumentVisibility.ADMIN, TeamMemberRole.ADMIN], () => true)
+      .otherwise(() => false);
+  }
+
   if (!document) {
     redirect(documentRootPath);
   }
 
+  if (!canAccessDocument && team) {
```

Comment:
should I leave this code block here just in case? at the moment, if the user doesn't have access to the document, it's going to be redirected to the document root path -> `if (!document) { redirect(documentRootPath); }`